controllers/authController.js

```diff
@@ -0,0 +1,143 @@
+const { v4: uuidv4 } = require('uuid');
+const passport = require('../config/passport.js');
+
+const User = require('../models/user.js');
+const jwt = require('jsonwebtoken');
+
+const signToken = payload =>
+  jwt.sign(payload, process.env.JWT_ACCESS_SECRET, {
+    expiresIn: process.env.JWT_ACCESS_EXPIRE_TIME,
+  });
+
+const generateRefreshToken = userId => {
+    return jwt.sign({ id: userId }, process.env.JWT_REFRESH_SECRET, { 
+    expiresIn: process.env.JWT_REFRESH_EXPIRE_TIME,
+  });
+};
+
+const auth = async (req, res, next) => {
+  await passport.authenticate('jwt', { session: false }, async (err, user) => {
+    if (!user || err) {
+      return res.status(401).json({
+        status: 'fail',
+        message: 'Unauthorized',
+      });
+    }
+
+    req.user = user;
+    next();
+  })(req, res, next);
+};
+
+const register = async (req, res, next) => {
+    try {
+      const { body } = req;
+      const { email, password } = body;
+  
+    const user = await User.create({
+        email,
+        password,
+        verificationToken: uuidv4(),
+      });
+       
+    res.status(201).json({
+        status: 'success',
+        message:
+          'Account created.',
+      });
+    } catch (err) {
+      res.status(400).json({ status: 'fail', message: err.message });
+    }
+  };
+
+  const login = async (req, res, next) => {
+    try {
+      const { email, password } = req.body;
+  
+      if (!email || !password)
+        return res
+          .status(400)
+          .json({ status: 'fail', message: 'Please provide an email or password' });
+  
+      const user = await User.findOne({
+        email,
+      }).select('password email verify name');
+  
+      if (!user || !(await user.isCorrectPassword(password, user.password)))
+        return res
+          .status(400)
+          .json({ status: 'fail', message: 'The email or password is incorrect!' });
+  
```

Comment:
muszę poczytać odnośnie refresh tokenów ale aktualnie jest on do niczego bo tylko access pozwala na interakcję a przez to że masz kilkanaście s ustawione no to jest kicha